### PR TITLE
[ISSI] Support IS31FL3741 and IS31FL3741A

### PR DESCRIPTION
**Pull request by melgeek001365**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/rgb_matrix_drivers.c [hunk 1]
```diff
@@ -44,8 +46,10 @@ static void init(void) {
         IS31FL3731_set_led_control_register(index, enabled, enabled, enabled);
 #    elif defined(IS31FL3733)
         IS31FL3733_set_led_control_register(index, enabled, enabled, enabled);
-#    else
+#    elif defined(IS31FL3733)
```
Comment (drashna) [inline quantum/rgb_matrix_drivers.c:49]: Whoops, I think you meant: 
```suggestion
#    elif defined(IS31FL3737)
```

**Comment by yiancar** [comment]
the actual driver looks good to me, but I dont have the chip at hand to try it

**Comment by melgeek001365** [comment]
> the actual driver looks good to me, but I dont have the chip at hand to try it

Haha. The 3741 chip has new features than 3733 and 3737. I have tested it on my board.

File: quantum/rgb_matrix_drivers.c (hunk 1 above)
Comment (melgeek001365) [reply quantum/rgb_matrix_drivers.c]: Thank you man!

**Comment by melgeek001365** [comment]
> the actual driver looks good to me, but I dont have the chip at hand to try it

File: quantum/rgb_matrix_drivers.c [hunk 2]
```diff
@@ -23,7 +23,11 @@
  * be here if shared between boards.
  */
 
-#if defined(IS31FL3731) || defined(IS31FL3733) || defined(IS31FL3737)
+<<<<<<< HEAD
+#if defined(IS31FL3731) || defined(IS31FL3733) || defined(IS31FL3737) || defined(IS31FL3737) || defined(IS31FL3741) 
+=======
+#if defined(IS31FL3731) || defined(IS31FL3733) || defined(IS31FL3737) || defined(IS31FL3741) 
+>>>>>>> 453fc33066a0205bce25d10bb54338cda8bf1efd
```
Comment (tzarc) [inline quantum/rgb_matrix_drivers.c:30]: Unresolved conflict here.

**Comment by tzarc** [comment]
I'd also imagine that the Z70Ultra keyboard needs to be put through a separate PR. I believe we'd normally keep things separate for historical purposes.

File: keyboards/melgeek/z70ultra/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+			                                                                   KC_HOME,  KC_PGUP,       KC_UP,  KC_PGDN,  KC_END,  KC_LEFT,  KC_DOWN, KC_RIGHT),
+		[1] = LAYOUT( /* FN */
+		    KC_GRV,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,            KC_F11,   KC_F12,   _______, \
+		    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,           _______,  _______,  RESET  , \
+		    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,           _______,            _______, \
+		    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,           _______,  _______,           \
+			_______,            _______,            _______,  _______,  _______,                      _______,  _______,                                        \
+			                                                                      _______,  _______,  _______,  _______,  _______, _______,  _______,  _______),
```
Comment (Erovia) [inline keyboards/melgeek/z70ultra/keymaps/via/keymap.c:17]: VIA defaults to 4 layers.
Please add 2 more empty layers.

File: drivers/issi/is31fl3741.c [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    IS31FL3741_write_register(addr, ISSI_REG_PULLDOWNUP, 0x77);
+
+// IS31FL3741_update_led_scaling_registers(addr, 0xFF, 0xFF, 0xFF);
+
+// Wait 10ms to ensure the device has woken up.
+#ifdef __AVR__
+    _delay_ms(10);
+#else
+    wait_ms(10);
+#endif
```
Comment (fauxpark) [inline drivers/issi/is31fl3741.c:195]:
```suggestion
    wait_ms(10);
```

File: drivers/issi/is31fl3741.c [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef __AVR__
+#    include <avr/interrupt.h>
+#    include <avr/io.h>
+#    include <util/delay.h>
+#else
+#    include "wait.h"
+#endif
```
Comment (fauxpark) [inline drivers/issi/is31fl3741.c:26]:
```suggestion
#include "wait.h"
```

File: drivers/issi/is31fl3741.c [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+
+#include "is31fl3741.h"
+#include <string.h>
+#include "i2c_master.h"
+#include "progmem.h"
+
+#include <debug.h>
```
Comment (fauxpark) [inline drivers/issi/is31fl3741.c:33]:
```suggestion
```

File: drivers/issi/is31fl3741.h [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef IS31FL3741_DRIVER_H
+#define IS31FL3741_DRIVER_H
```
Comment (fauxpark) [inline drivers/issi/is31fl3741.h:21]:
```suggestion
#pragma once
```

File: drivers/issi/is31fl3741.h [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+// (eg. from a timer interrupt).
+// Call this while idle (in between matrix scans).
+// If the buffer is dirty, it will update the driver with the buffer.
+void IS31FL3741_update_pwm_buffers(uint8_t addr1, uint8_t addr2);
+void IS31FL3741_update_led_control_registers(uint8_t addr1, uint8_t addr2);
+void IS31FL3741_set_scaling_registers(const is31_led *pled, uint8_t red, uint8_t green, uint8_t blue);
+
+void IS31FL3741_set_pwm_buffer(const is31_led *pled, uint8_t red, uint8_t green, uint8_t blue);
+
+#endif  // IS31FL3741_DRIVER_H
```
Comment (fauxpark) [inline drivers/issi/is31fl3741.h:58]:
```suggestion
```

**Comment by melgeek001365** [comment]
> separate

Yes. It make sense. I will try it.

**Pull request closed without merging**